### DISCUSSION TITLE
Will Home Assistant Amber be supported?

### DISCUSSION THREAD

**Discussion by osos** [Koenkk/zigbee2mqtt#8766] [General]
Incoming Home Assistant Amber with built-in Zigbee. https://www.crowdsupply.com/nabu-casa/home-assistant-amber

Will it be supported by Zigbee2Mqtt ?

**Comment by Hedda** [comment]
Was about to post about "Home Assistant Amber" Raspberry Pi CM4 SBC carrier board with onboard Silicon Labs EZSP v8 Zigbee RF module.

zigbee-herdsman (and thus Zigbee2MQTT and IoBroker which depends on zigbee-herdsman) do already have initial experimental support for Silabs EZSP (EmberZNet Serial Protocol), but as far as I know, no developer is currently actively working on improving support for EZSP.

https://www.zigbee2mqtt.io/information/supported_adapters.html

You can find much more information and discussion about zigbee-herdsman development on EFR32 EmberZNet EZSP adapter support here:

https://github.com/Koenkk/zigbee-herdsman/issues/319

https://github.com/Koenkk/zigbee-herdsman/pull/317

https://github.com/Koenkk/zigbee-herdsman/pulls?q=ezsp

FYI, for those who missed this news about Home Assistant Amber:

Home Assistant has just announced an official hardware crowdfunding campaign for a "Home Assistant Amber" Raspberry Pi Compute Module 4 single-board computer carrier board with an integrated Silicon Labs MGM210P (EFR32MG21) based Zigbee radio module with a ceramic chip antenna and hardware flow control pins connected, pre-flashed with Silabs EmberZNet firmware and an unlocked bootloader (plus even a debug header for the EFR32 module), as well as a 1GbE PoE port and NVMe / M.2 expansion port for an SSD or AI accelerator.

https://www.home-assistant.io/blog/2021/09/13/home-assistant-amber/

https://www.crowdsupply.com/nabu-casa/home-assistant-amber

https://www.youtube.com/watch?v=KPR-shzT_VQ

![image](https://user-images.githubusercontent.com/6320001/133604717-8372f68d-f526-4918-bc40-a9f314d849fb.png)

Could work as a nice Silabs EFR32 development kit with the debug header and unlocked bootloader.

https://www.cnx-software.com/2021/09/16/home-assistant-amber-raspberry-pi-cm4-home-automation-gateway/

https://linuxgizmos.com/home-assistant-project-launches-its-own-raspberry-pi-cm4-carrier/

![image](https://user-images.githubusercontent.com/6320001/133620084-88eeef6d-e37c-469a-89b2-f27069708842.png)

They noted they are a participant project of Silicon Labs IoT Accelerator cross-promotion with Mouser Electronics and Crowd Supply:

https://www.crowdsupply.com/silicon-labs/iot-accelerator

PS: They also noted this is the successor to Home Assistant Blue which has been discontinued though they will continue to support it.

https://www.cnx-software.com/2020/12/16/odroid-n2-based-home-assistant-blue-announced-as-official-hardware-for-home-assistant/

**Reply by Hedda** [reply]
> Home Assistant has just announced an official hardware crowdfunding campaign for a "Home Assistant Amber" Raspberry Pi Compute Module 4 single-board computer carrier board with an integrated Silicon Labs MGM210P (EFR32MG21) based Zigbee radio module with a ceramic chip antenna and **hardware flow control pins connected** plus pre-flashed with Silabs EmberZNet firmware, as well as a 1GbE PoE port and NVMe / M.2 expansion port for an SSD or AI accelerator.

One question I have is if zigbee-herdsman / Zigbee2MQTT will work with this EZSP module when hardware flow control is enabled?

RTS / CTS Hardware Flow Control should in theory offer more stable/reliable communication with this UART module.

**Comment by Hedda** [comment]
You might want to consider Electrolama zoe2 (TI CC1352P Zigbee with PoE + RTC shield for RPi) as an alternative to "Home Assistant Amber":

https://electrolama.com/projects/zoe2/

zoe2 - CC1352P Radio + RTC + PoE for Raspberry Pi

https://shop.electrolama.com/collections/raspberry-pi-addons/products/zoe2-cc1352p-radio-rtc-poe-for-raspberry-pi?variant=41086023008417

https://shop.electrolama.com/collections/raspberry-pi-addons/products/zoe2-cc1352p-radio-rtc-poe-for-raspberry-pi?variant=37646454161569

**Reply by boesing** [reply]
Oh I'd love to but this has no support for nvme SSD 😔

**Reply by Hedda** [reply]
> Oh I'd love to but this has no support for nvme SSD 😔

No, but the Raspberry Pi 4 has USB 3.0 ports so a standard 2,5-inch SSD SATA disk in a USB enclosure should be less expensive than a NvME disk, (just make sure that the USB enclosure and the cable to it are shielded from EMF or use a long antenna to the Zigbee radio).

If going to run it as a dedicated Home Assistant OS box then doubt that the Home Assistant application itself will benefit much from NvME disk speeds on PCIe x1 (1 PCI lane) versus running on a standard 2,5-inch SSD SATA disk in a USB enclosure. Personally, I would prefer using it over a classic 2,5-inch SATA harddrive with a spinning-disk on a USB 3.0 as speed is good enough and have a much higher write cycle lifetime so should last many more years if buy a quality disk.

Another benefit with Raspberry Pi 4 and its USB 3.0 ports is that can use a Google Coral USB Accelerator over USB 3.0 speeds. As I understand it the Home Assistant Amber only has USB 2.0 ports and the Google Coral USB Accelerator adapter is USB 3.0 (not USB 2.0) so it is not clear that can really use a Google Coral USB Accelerator with it over USB 2.0 ports only, ...and we know that they did not actually test Google Coral M.2 AI Accelerator before saying it would work which turns out to be not true, so not sure if they really tested it with Google Coral USB Accelerator adapter over USB 2.0 speeds or not:

https://www.crowdsupply.com/nabu-casa/home-assistant-amber/updates/thank-you-video-demo-and-google-coral

**Reply by boesing** [reply]
I already have an NVMe which is attached via USB but I'd love to have anything in-place.
The only reason why I would not buy the crowdfunded amber platine is due to the fact that it is not properly supported by zigbee2mqtt which is a showstopper. Zigbee is quite essential for my home automation and thus I want to stick to zigbee2mqtt which I probably already supported with some money over the last 12 months because I love this project.

I really like the way how the amber project integrates anything into one single board (besides the fact that they use that zigbee controlelr). I don't care about full NVMe speed but about how it integrates. Having external enclosures for either 2.5" or NVMe sucks (as I do already have that) and thus having a way to attach my SSD directly to the board would be awesome.

So, even tho 2.5" would cost less, that size is outdated and I'd rather stick with NVMe.

**Reply by Hedda** [reply]
FYI, there's also many “set-top box” style enclosures that feature integrated space / internal bay for M.2/NVME or 2.5-inch HDD. Ex:

https://www.argon40.com/argon-one-m-2-case-for-raspberry-pi-4.html

https://www.argon40.com/argon-one-v-2-case-for-raspberry-pi-5.html + https://www.argon40.com/argon-one-m-2-expansion-board.html

https://www.cnx-software.com/2020/09/25/deskpi-pro-raspberry-pi-4-case-comes-with-2-5-inch-hdd-ssd-bay-full-sized-hdmi-ports-pwm-fansink/

https://thepihut.com/collections/raspberry-pi-ssd-cases/products/raspberry-pi-quattro-case-with-vesa-mount

https://thepihut.com/collections/raspberry-pi-ssd-cases/products/retroflag-nespi-4-case-for-raspberry-pi-4

If you do not need PoE then there are now also more than one Raspberry Pi GPIO HAT/shield based on TI's CC2652/CC1352 chips:

https://www.zigbee2mqtt.io/guide/adapters/#supported-adapters

https://zig-star.com/projects/zigbee-shield/

https://shop.codm.de/automation/zigbee/33/zigbee-cc2652p2-raspberry-pi-module

https://electrolama.com/projects/zoe2/ "zoe2-lite" = CC1352P Radio for Raspberry Pi (Radio only)

**Reply by boesing** [reply]
I will have a look on these. I would prefer something which is compatible with the compute model 4 as the availability of the regular Pi ain't good. Could get one for 125 EUR (8GB) on Amazon but could get a compute model 4 with 8GB and 16GB storage for around 86 EUR.

Haven't spent too much time on that but PoE via Unifi switch would be awesome but I am not even aware if that power would be enough to handle a Pi and a NVMe SSD 😅

**Comment by Hedda** [comment]
Note! EZSP support is still in a very experimental phase so you should really post your feedback here -> https://github.com/Koenkk/zigbee-herdsman/issues/319

I am not sure if developer(s) is currently actively working on improving or maintaining that experimental support(?).

Only clear tips is to try upgrade EmberZNet NCP UART EZSP firmware Zigbee firmware version 6.7.9 or later and test dev branch (a.k.a. Edge):

https://github.com/arendst/Tasmota/tree/development/tools/fw_SonoffZigbeeBridge_ezsp

https://github.com/Koenkk/zigbee2mqtt/tree/dev

There looks to be some ezsp related fixes in the zigbee-herdsman dependency that might have not yet made it into Z2M dev/edge yet:

https://github.com/Koenkk/zigbee-herdsman/

Regardless you still want to direct most if not all feedback and issues to subscribers of the WIP discussion in the zigbee-herdsman repo:

https://github.com/Koenkk/zigbee-herdsman/issues/319

Also, please take note that WiFi based Zigbee bridges like ITead Sonoff ZBBridge are generally not recommended with the EZSP protocol. See the warnings about it:

https://www.zigbee2mqtt.io/guide/adapters/#notes

WiFi-based Serial-to-IP bridges are not recommended for Silicon Labs EZSP adapters as the serial protocol does not have enough fault-tolerance to handle packet loss or latency delays that can normally occur over WiFi connections. If cannot use a locally connected USB or UART/GPIO adapter then the recommendation is to use remote adapter that connected via Ethernet (wired) to avoid issues with EZSP caused by WiFi connections.

https://www.zigbee2mqtt.io/advanced/remote-adapter/connect_to_a_remote_adapter.html

WiFi-based Serial-to-IP bridges are not recommended for Silicon Labs EZSP adapters as the serial protocol does not have enough fault-tolerance to handle packet loss or latency delays that can normally occur over WiFi connections.

More warning about Zigbee to WiFi bridges for EZSP is not recommended for production as it does not do well with packed loss:

https://github.com/zigpy/bellows#warning-about-zigbee-to-wifi-bridges

https://www.home-assistant.io/integrations/zha/#warning-about-wi-fi-based-zigbee-to-serial-bridgesgateways

FYI, I also understand that the EFR32 board antenna design of the "SM-011 V1.0" Zigbee module used inside ITead's Sonoff ZBBridge have very high receive sensitivity to electromagnetic interference (EMI) / radio-frequency interference. This means that it will be extra important to keep it away from any other electronic devices or appliances that may generate an electromagnetic field (EM Field) or send our interfering radio signals. That includes the computer that you run your home automation application software on as well as any devices such as for example external harddrives or storage devices, power-supplies, etc., and of course any Wi-Fi access points or routers nearby.

Note! The new ITead Zigbee 3.0 USB Dongle Model: 9888010100045 (Hardware Revision Version 1.3) circuit board share the same type of antenna design so its radio is as well extremely sensitive to signal interference on its receiving end. The current recommendation for using it is to connect it via a long USB extension cable and try to move it as far away from all electronics and other possible sources of signal interference.

PS: Another general tip that always applies when building a Zigbee network is to begin by adding a few good mains-powered Zigbee router devices so that they will make use a stable baseline. Many start by buying three "IKEA TRADFRI Signal Repeater" (E1746).

https://zigbee.blakadder.com/Ikea_E1746.html

https://www.ikea.com/us/en/p/tradfri-signal-repeater-30400407/

https://www.ikea.com/gb/en/p/tradfri-signal-repeater-80424255/

https://www.ikea.com/se/sv/p/tradfri-signalfoerstaerkare-10400408/
